ext/calendar/calendar.c

```diff
@@ -131,8 +131,10 @@ static void _php_cal_info(int cal, zval *ret)
 	calendar = &cal_conversion_table[cal];
 	array_init(ret);
 
-	array_init(&months);
-	array_init(&smonths);
+	array_init_size(&months, calendar->num_months + 1);
+	array_init_size(&smonths, calendar->num_months + 1);
```

Comment:
Note that the loop starts at 1.

```suggestion
	array_init_size(&months, calendar->num_months);
	array_init_size(&smonths, calendar->num_months);
```

Reply:
ah yes month number ...

Reply:
Ah actually I might've f-ed up here, as the indices added to the array are also 1-based, so you need the extra "empty" array slot. Sorry